From ac90985a323a376b8567710105fdb01e6102e6be Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 19 Jan 2018 01:33:11 +0000
Subject: [PATCH] update-kernel: handle -vanilla suffix in System.map, config
 and vmlinuz

Previously the vanilla kernel did not add any suffix to System.map,
config and vmlinuz for the vanilla kernel, but it does now. Fix
update-kernel to handle both cases.

Note also that /lib/modules/$ver still does not have any -vanilla
suffix.
---
 update-kernel.in | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/update-kernel.in b/update-kernel.in
index 88f266b..930b3eb 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -254,7 +254,11 @@ mkinitfs $MKINITFS_ARGS -q -b $ROOT -F "$features base squashfs" \
 	-o "$STAGING/initramfs-$FLAVOR" "$KVER"
 
 for file in System.map config vmlinuz; do
-	cp "$BOOT/$file$KVER_FLAVOR" $STAGING
+	if [ -f "$BOOT/$file-$FLAVOR" ]; then
+		cp "$BOOT/$file-$FLAVOR" $STAGING
+	else
+		cp "$BOOT/$file" $STAGING
+	fi
 done
 
 if [ "$MNTDIR" ]; then
-- 
2.15.0

